FWIW, here is how things eventually fail here (x86_64 system), running `nix-build -E 'let pkgs = import ./. { overlays = []; }; in pkgs.pkgsi686Linux.openjdk15` while at the root of nixpkgs at commit be4763a696daf2fcb75b91aa4b8533257e466579 (current tip of your branche) — similar things happen when I try `pkgsi686Linux.nixosTests.gerrit`, which appears to have the same list of to-build derivations:

```
g++ -std=gnu++98 -fno-PIE -c   -g -DIN_GCC     -fno-exceptions -fno-rtti -fasynchronous-unwind-tables -W -Wall -Wno-narrowing -Wwrite-strings -Wcast-qual -Wno-format -Wmissing-format-attribute -Woverloaded-virtual -pedantic -Wno-long-long -Wno-variadic-macros -Wno-overlength-strings -fno-common  -DHAVE_CONFIG_H -I. -I. -I../../gcc-8.4.0/gcc -I../../gcc-8.4.0/gcc/. -I../../gcc-8.4.0/gcc/../include -I../../gcc-8.4.0/gcc/../libcpp/include -I/nix/store/cjavgfszkgqi323khmil471q7ixk19p3-gmp-6.2.1-dev/include -I/nix/store/xl28qirdv9z6wvv8grhryyay22hjnkpp-mpfr-4.1.0-dev/include -I/nix/store/klzzqwpms24jd778slylz0hjng0lqlkj-libmpc-1.2.0/include  -I../../gcc-8.4.0/gcc/../libdecnumber -I../../gcc-8.4.0/gcc/../libdecnumber/bid -I../libdecnumber -I../../gcc-8.4.0/gcc/../libbacktrace -I/nix/store/bcsl8r9pzqiwk2959kaszdqj4dwsckyw-isl-0.17.1/include  -o ree.o -MT ree.o -MMD -MP -MF ./.deps/ree.TPo ../../gcc-8.4.0/gcc/ree.c

ERROR: Build failed for target 'all' in configuration 'linux-x86-normal-server-release' (exit code 2)
Stopping sjavac server
g++ -std=gnu++98 -fno-PIE -c   -g -DIN_GCC     -fno-exceptions -fno-rtti -fasynchronous-unwind-tables -W -Wall -Wno-narrowing -Wwrite-strings -Wcast-qual -Wno-format -Wmissing-format-attribute -Woverloaded-virtual -pedantic -Wno-long-long -Wno-variadic-macros -Wno-overlength-strings -fno-common  -DHAVE_CONFIG_H -I. -I. -I../../gcc-8.4.0/gcc -I../../gcc-8.4.0/gcc/. -I../../gcc-8.4.0/gcc/../include -I../../gcc-8.4.0/gcc/../libcpp/include -I/nix/store/cjavgfszkgqi323khmil471q7ixk19p3-gmp-6.2.1-dev/include -I/nix/store/xl28qirdv9z6wvv8grhryyay22hjnkpp-mpfr-4.1.0-dev/include -I/nix/store/klzzqwpms24jd778slylz0hjng0lqlkj-libmpc-1.2.0/include  -I../../gcc-8.4.0/gcc/../libdecnumber -I../../gcc-8.4.0/gcc/../libdecnumber/bid -I../libdecnumber -I../../gcc-8.4.0/gcc/../libbacktrace -I/nix/store/bcsl8r9pzqiwk2959kaszdqj4dwsckyw-isl-0.17.1/include  -o inchash.o -MT inchash.o -MMD -MP -MF ./.deps/inchash.TPo ../../gcc-8.4.0/gcc/inchash.c

=== Output from failing command(s) repeated here ===
* For target buildtools_tools_jigsaw_classes__the.BUILD_JIGSAW_TOOLS_batch:
Connection attempt failed: Connection refused (Connection refused)
Connection attempt failed: Connection refused (Connection refused)
Connection attempt failed: Connection refused (Connection refused)
Giving up
IOException caught during compilation: Could not connect to server

* All command lines available in /build/source/build/linux-x86-normal-server-release/make-support/failure-logs.
=== End of repeated output ===

No indication of failed target found.
Hint: Try searching the build log for '] Error'.
Hint: See doc/building.html#troubleshooting for assistance.

make[1]: *** [/build/source/make/Init.gmk:305: main] Error 2
make: *** [/build/source/make/Init.gmk:186: all] Error 2
builder for '/nix/store/h79dsv5c8i12rpwkfk1h4nbgjb6ss20v-openjdk-headless-11.0.9+11.drv' failed with exit code 2
cannot build derivation '/nix/store/akfp4jrhw77ypm9dkk69990lk2f94j35-openjfx-modular-sdk-15.0.1+1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/jfdil4pribhqmpii5wkcnk8qcw3wnikx-openjdk-15.0.1-ga.drv': 1 dependencies couldn't be built
error: build of '/nix/store/jfdil4pribhqmpii5wkcnk8qcw3wnikx-openjdk-15.0.1-ga.drv' failed
```

So it actually fails building openjdk11. Maybe openjdk11 would require the same gcc8 override? That said, it could maybe also be due to my computer not having enough RAM to compile openjdk… though I haven't seen an OOM message in my logs so that'd be surprising.

Do you know of an x86_64 system where the above-listed nix-build command works?

--- a/pkgs/development/compilers/openjdk/openjfx/15.nix
+++ b/pkgs/development/compilers/openjdk/openjfx/15.nix
@@ -65,9 +65,7 @@ let
     # Downloaded AWT jars differ by platform.
     outputHash = {
       x86_64-linux = "0hmyr5nnjgwyw3fcwqf0crqg9lny27jfirycg3xmkzbcrwqd6qkw";
-      # The build-time dependencies don't currently build for i686, so no
-      # reason to fetch this one correctly either...
-      i686-linux = "0000000000000000000000000000000000000000000000000000";
+      i686-linux = "0hx69p2z96p7jbyq4r20jykkb8gx6r8q2cj7m30pldlsw3650bqx";
     }.${stdenv.system} or (throw "Unsupported platform");
   };
 

